thread-manager/README.md

```diff
@@ -0,0 +1,61 @@
+# thread-manager
+Balances machine resources between multiple threaded runtimes. The purpose is to manage thread contention
+between different parts of the code that may
+benefit from a diverse set of management options. For example, we may want to have cores 1-4 handling
+networking via Tokio, core 5 handling file IO via Tokio, cores 9-16 hallocated for Rayon thread pool,
+and cores 6-8 available for general use by std::thread. This will minimize contention for CPU caches
+and context switches that would occur if Rayon was entirely unaware it was running side-by-side with
+tokio, and each was to spawn as many threads as there are cores.
+
+# Supported threading models
+## Affinity
+All threading models allow setting core affinity, but only on linux
+
+For core affinity you can set e.g.
+```toml
+core_allocation.DedicatedCoreSet = { min = 16, max = 64 }
+```
+to pin the pool to cores 16-64.
+
+## Scheduling policy and priority
+If you want you can set thread scheduling policy and priority. Keep in mind that this will likely require
+```bash
+ sudo setcap cap_sys_nice+ep
```

Comment:
in the past we've had tuning scripts that could handle recommended configurations like this. If it makes sense, we could add this wherever relevant

Reply:
I'm not sure about priority stuff at this point, it is super platform specific and mostly a placeholder at the moment, as getting this past CI will be unfun (due to extra priveleges necessary)